tdms/tests/system/utils.py

```diff
@@ -67,7 +80,7 @@ def matches(self, other: "HDF5File", rtol=1e-10) -> bool:
         arrays
         """
 
-        for key, value in self.items():
+        for key, value in self.traverse(self):
```

Comment:
picky and annoying I know, and I don't have a good rational, but I don't like the pattern

```python
self.method(self)
```

I may be inclined to do `_traverse(self)`, but also 🤷🏼